File: VB/WinWebSolution.Module/InstantiateAggregatedPropertiesViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace WinWebSolution.Module
	Partial Public Class InstantiateAggregatedPropertiesViewController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
		End Sub
		Private Sub simpleAction1_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs) Handles simpleAction1.Execute
			Dim obj As TestAggregatedPropertiesObject = TryCast(View.CurrentObject, TestAggregatedPropertiesObject)
			If obj IsNot Nothing Then
				obj.SetReadOnlyAddress()
			End If
		End Sub
		Private Sub simpleAction2_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs) Handles simpleAction2.Execute
			Dim obj As TestAggregatedPropertiesObject = TryCast(View.CurrentObject, TestAggregatedPropertiesObject)
			If obj IsNot Nothing Then
				obj.SetWritableAddress()
			End If
		End Sub
	End Class
End Namespace

File: CS/E1284/WinWebSolution.Module/InstantiateAggregatedPropertiesViewController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace WinWebSolution.Module {
    public partial class InstantiateAggregatedPropertiesViewController : ViewController {
        public InstantiateAggregatedPropertiesViewController() {
            InitializeComponent();
            RegisterActions(components);
        }
        private void simpleAction1_Execute(object sender, SimpleActionExecuteEventArgs e) {
            TestAggregatedPropertiesObject obj = View.CurrentObject as TestAggregatedPropertiesObject;
            if (obj != null) {
                obj.SetReadOnlyAddress();
            }
        }
        private void simpleAction2_Execute(object sender, SimpleActionExecuteEventArgs e) {
            TestAggregatedPropertiesObject obj = View.CurrentObject as TestAggregatedPropertiesObject;
            if (obj != null) {
                obj.SetWritableAddress();
            }
        }
    }
}